activation/nipost.go

```diff
@@ -388,23 +418,27 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 	})
 }
 
-// Submit the challenge to all registered PoETs.
-func (nb *NIPostBuilder) submitPoetChallenges(
+// Submit the challenge to registered PoETs with given addresses.
+func (nb *NIPostBuilder) submitPoetChallengesToAddresses(
 	ctx context.Context,
 	signer *signing.EdSigner,
 	deadline time.Time,
+	addresses map[string]bool,
```

Comment:
How about using a `struct{}` for the value type if you need a _set_?
```suggestion
	addresses map[string]struct{},
```